src/Middleware/ImplicitHeadMiddleware.php

```diff
@@ -47,28 +48,24 @@ class ImplicitHeadMiddleware implements MiddlewareInterface
     public const FORWARDED_HTTP_METHOD_ATTRIBUTE = 'forwarded_http_method';
 
     /**
-     * @var callable
+     * @var RouterInterface
      */
-    private $responseFactory;
+    private $router;
 
     /**
      * @var callable
      */
     private $streamFactory;
 
     /**
-     * @param callable $responseFactory A factory capable of returning an
-     *     empty ResponseInterface instance to return for implicit HEAD
-     *     requests.
      * @param callable $streamFactory A factory capable of returning an empty
      *     StreamInterface instance to inject in a response.
      */
-    public function __construct(callable $responseFactory, callable $streamFactory)
+    public function __construct(RouterInterface $router, callable $streamFactory)
```

Comment:
Another note for when we backport: this is a signature change for the constructor. As long as we provide a factory, we should be fine, but we may need the version in zend-expressive to remain untouched, and note that developers should opt-in to the version in the zend-expressive-router package.